ci/travis_before_script_cpp.sh

```diff
@@ -51,11 +53,12 @@ if [ "$ARROW_TRAVIS_USE_TOOLCHAIN" == "1" ]; then
   # HACK(wesm): We started experiencing OpenSSL failures when Miniconda was
   # updated sometime on October 2 or October 3
   conda update -y -p $CPP_TOOLCHAIN ca-certificates -c defaults
+else
+  if [ $TRAVIS_OS_NAME == "osx" ]; then
+    brew update && brew bundle -v --file=cpp/Brewfile
```

Comment:
Can we use conda for whatever's in this file and forego the use of brew altogether?

Reply:
This depends on if the `c_glib` build can be done using the conda artifacts. I will have a look into this later.

Reply:
Weirdly the `c_glib` build finished in 16min overall despite the use of brew. This may simply have been a problem in the combination of conda+brew.